packages/quantic/force-app/main/default/lwc/quanticStandaloneSearchBox/quanticStandaloneSearchBox.js

```diff
@@ -232,21 +224,35 @@ export default class QuanticStandaloneSearchBox extends NavigationMixin(
 
   clearInput() {
     this.input.value = '';
-    this.actualSearchBox.updateText(this.input.value);
+    this.standaloneSearchBox.updateText(this.input.value);
     this.input.focus();
   }
 
   handleSuggestionSelection(event) {
     const textValue = event.detail;
-    this.actualSearchBox.selectSuggestion(textValue);
+    this.standaloneSearchBox.selectSuggestion(textValue);
+  }
+
+  resetSearchboxState(engineId) {
+    const engine = getHeadlessBindings(engineId)?.engine;
+    if(!engine) {
+      return;
+    }
+    const {updateQuery} = CoveoHeadless.loadQueryActions(engine);
+
+    engine.dispatch(updateQuery({q: ''}));
   }
 
   navigateToSearchPage() {
+    const value = this.standaloneSearchBox.state.value;
+
+    this.resetSearchboxState(this.engineId);
+    this.resetSearchboxState(this.standaloneEngineId);
     this[NavigationMixin.Navigate](
       {
         type: 'standard__webPage',
         attributes: {
-          url: `${this.redirectUrl}#q=${this.standaloneSearchBox.state.value}`,
+          url: `${this.redirectUrl}#q=${value}`,
```

Comment:
I know that some client like include stuff in the url hash do you think we might want to consider building the url using URL api instead of using string ? https://developer.mozilla.org/en-US/docs/Web/API/URL_API

Reply:
Good point. I'll check it out

Reply:
Didn't you mention there was an issue with the `#` whenever the query was a `#` for example with the urlManager?

Reply:
I believe that's fixed now. I can't reproduce.

Reply:
Ask Lucille :D

Reply:
😱